src/core/utils/DenoiserManager.js

```diff
@@ -0,0 +1,208 @@
+			map: null,
+			transparent: true,
+			blending: NoBlending,
+
+			premultipliedAlpha: renderer.getContextAttributes().premultipliedAlpha,
+		} );
+		this.ptMaterial.opacity = 1;
+
+		// Material to blend between pathtracer and denoiser
+		// TODO: Not sure if  params are required
+		this.blendMaterial = new BlendMaterial();
+
+		this.quad = new FullScreenQuad( this.ptMaterial );
+		this.createConversionRenderTarget( renderer.domElement.width, renderer.domElement.height );
+
+	}
+
+	async denoise( rawPathtracedTexture, albedoTexture, normalTexture ) {
+
+		this.isDenoising = true;
+		// Adjust the height /width if changed from before
+		const height = this.renderer.domElement.height;
+		const width = this.renderer.domElement.width;
+
+		if ( this.denoiser.height !== height || this.denoiser.width !== width ) {
+
+			this.denoiser.width = width;
+			this.denoiser.height = height;
+			this.createConversionRenderTarget( width, height );
+
+			/* Used when debugging
+			if ( this.rawCanvas ) {
+
+				this.rawCanvas.width = width;
+				this.rawCanvas.height = height;
+
+			}*/
+
+		}
+
+		// set so we can access later when blending
+		this.pathtracedTexture = this.getCorrectPathtracerTexture( rawPathtracedTexture );
+		const colorWebGLTexture = this.getWebGLTexture( this.pathtracedTexture );
+		const albedoWebGLTexture = this.getWebGLTexture( albedoTexture );
+		const normalWebGLTexture = this.getWebGLTexture( normalTexture );
+
+		//* run the denoiser
+		this.renderer.resetState();
+
+		/* The setting of inputs is async and running too quick before execute. this will be fixed in a future version
+		see https://github.com/DennisSmolek/Denoiser/issues/21
+		const denoisedWebGLTexture = await this.denoiser.execute( colorWebGLTexture, albedoWebGLTexture, normalWebGLTexture );
+		*/
+
+		// Set input Textures with await
+		await this.denoiser.setInputTexture( 'color', colorWebGLTexture );
+		if ( albedoTexture ) await this.denoiser.setInputTexture( 'albedo', albedoWebGLTexture );
+		if ( normalTexture ) await this.denoiser.setInputTexture( 'normal', normalWebGLTexture );
+		// Run the denoiser
+		const denoisedWebGLTexture = await this.denoiser.execute();
+
+		this.renderer.resetState();
+		if ( ! this.outputTexture ) this.outputTexture = this.createOutputTexture();
+		// inject the webGLTexture into the texture
+		this.denoisedTexture = this.injectWebGLTexture( this.outputTexture, denoisedWebGLTexture );
+		// mark as complete and setup the renderer
+		this.isDenoising = false;
+		this.denoiserFinished = performance.now();
+		return this.denoisedTexture;
+
+	}
+
+	// render the blended output
+	renderOutput( bypassTexture ) {
+
+		if ( ! this.pathtracedTexture || ! this.denoisedTexture ) return;
+
+		//const currentTarget = this.renderer.getRenderTarget();
+		this.quad.material = this.blendMaterial;
+		this.blendMaterial.target1 = this.pathtracedTexture;
+		this.blendMaterial.target2 = this.denoisedTexture;
+		this.blendMaterial.t2conversion = false;
```

Comment:
When testing against this PR, I found that the results were coming out too dark. Due to the colorspace issue, it looks like you are generating t2conversion uniform. When outputting denoised results, it looks fine if you set t2 conversion to true. Please confirm the fix.

Reply:
I too saw the brighter output, but note that the colors are much flatter.
To test, set the samples really high, like 100. Use that as your Ground Truth. You'll see the results are somewhere between the current output and the converted output.

the t2conversion was my original solution and may be the correct way to go, however, the conversion earlier in the pipeline may yield better results.

Should we try converting the raw pathtracer output before it goes into the denoiser? If we do that we must also convert the albedo pass.

I'm willing to try whichever ways people think best or even add debug options in them so we can test. 

I figured it best to bring out and let others try and see what everyone thinks the best approach is.

Reply:
When I was testing OIDN, all inputs had to be set to SRGB colorspace (pathtraced, albedo, normal), and when displaying the denoised result, there was no color distortion when outputting the denoised result without changing the colorspace. Let's run a related test.

Reply:
[The OIDN docs](https://github.com/RenderKit/oidn?tab=readme-ov-file#rt) say:
>  srgb | false | the main input image is encoded with the sRGB (or 2.2 gamma) curve (LDR only) or is linear; the output will be encoded with the same curve

Which says they are linear by default and can be SRGB.
[Later](https://github.com/RenderKit/oidn?tab=readme-ov-file#rt) they say the model is trained in srgb space so I agree that we should try to keep input as SRGB.

In the render targets I set the colorspace to `THREE.SRGBColorSpace`. Shouldn't that make the renderer output to that colorspace?

This is confusing because the `t2Conversion` is `LinearToSRGB()`

Reply:
I noticed the colors being significantly darker, as well - it does look like a color space issue. I can help with that later on.

| raw | denoised |
|---|---|
| ![image](https://github.com/user-attachments/assets/cad65c42-5876-43f4-b02d-21fd5ae900cc) | ![image](https://github.com/user-attachments/assets/da37221f-0bff-478f-9fed-cf652ab4f744) |

Reply:
Colors look a look a lot nicer with `t2conversion = true` but we have banding issues (see lighter color at the top) because we're storing linear srgb colors in 8 bits, causing precision loss. I'll trace the color space of the textures once some of the other changes are made. The quality is looking nice, though!

<img width="600" alt="image" src="https://github.com/user-attachments/assets/0b935fec-b634-4b85-be47-3d321aa6f193">

Reply:
After some battling I've worked this out. Also the denoiser hates the gradient background